### ISSUE TITLE
https://www.gmx.net/ and https://web.de/ broken

### ISSUE THREAD

**Issue by akks1981** [uBlockOrigin/uAssets#7508]
###` https://www.gmx.net/` and `https://web.de/`

### Describe the issue

if ublock is switched on, nothing is shown on the pages. If ublock is switched off everything is displayed normally.

### Versions

- Browser/version: Firefox 78b2
- uBlock Origin version: 1.27.11b2

### Settings

standart

**Comment by akks1981** [comment]
now it works....dk why

**Comment by webfussel** [comment]
Hey everyone,

I'm a software architect at WEB.DE/GMX and I was looking into the problem.
On the day this problem occured we released a new version of given websites.
After we realized the page appeared blank when loading under certain circumstances we rolled back our release as fast as possible.
The main problem seemed to be "uBlock".

Now the problem is:
There's a custom filter that blocks the basic JavaScript functionality of "CustomEvent" for WEB.DE/GMX, which is unfortunate in different ways.
It was put on the list some time in November 2019 based on video ads we have.


Blocking "CustomEvent" is a rather big issue for us.
We had some code refactorings in the past - before the CustomEvent for us was put on the EasyPrivacy List - to upgrade to newer technologies
and now rely heavily on this functionality for our site to work.
This includes some layouting techniques as well as our cookie banner.
Not being able to show the cookie banner can bring us pretty big legal problems.

The EU of course demands us to get user consent for cookie creation and usage.

We rolled back to not exclude uBlock customers from the best experience they can have on our page, but we decided to give you
four weeks to resolve this issue.
Of course we would be happy to offer coorporation on this topic, but after the set deadline of four weeks we have to roll out our
changes.

In the end we want to make sure also uBlock users can check their mails without problems!

Thank you.

Best regards,
Fiona

PS: Should I open a new issue for this topic?

**Comment by mapx-** [comment]
> before the CustomEvent for us was put on the EasyPrivacy List
I get no `CustomEvent` in `EasyPrivacy` list

> we decided to give you four weeks to resolve this issue

You should provide test pages with your future changes (including videos / video ads)

**Comment by okiehsch** [comment]
I am pretty sure she means our filter 
`##+js(set, CustomEvent, noopFunc)`

**Comment by mapx-** [comment]
Yes, but she was saying something about EP too

**Comment by okiehsch** [comment]
If I remove our filter I can see the cookie banner, so I think that's the main issue.
I

**Comment by mapx-** [comment]
We should wait their new pages / design / refactoring / see what's new about their video ads

**Comment by llacb47** [comment]
Maybe same redesign is coming to `https://www.mail.com/` as well?

**Comment by webfussel** [comment]
Hey everyone!

First of all: Thanks for reopening the issue so fast and looking into it :)

@mapx-: True. In the text I wrote EasyPrivacy and I have to admit: sorry, me and my proofreaders messed up here. I will edit the post and add a note that we actually meant the Filter **okiehsch** talked about. Sorry for the inconvenience on this point.

At the moment we use **CustomEvent** in the page itself for some resizing issues we faced and our Cookie Banner also uses it internally.

Main problem is, as okiehsch told, the Cookie Banner which must not be blocked.

At the moment this issue was opened we had a bug we wanted to resolve with a Custom Event. When we got the message ublock users can't see the page anymore we rolled back immediately - but we have to solve this issue soon.

You can test the current behaviour directly on live:
- [WEB.DE](https://web.de/)
- [GMX](https://www.gmx.net/)

What does not work:
- Resizing the page
- Cookie banner is not shown

@llacb47 I have no information about that, sorry!

**Comment by okiehsch** [comment]
https://github.com/uBlockOrigin/uAssets/commit/32f9a6b14694a2e3d818626b718dbf10fe62ad2e

I have removed the CustomEvent filter.

**Comment by JobcenterTycoon** [comment]
Video disappear for me without the CustomEvent filter (firefox)
`https://web.de/magazine/unterhaltung/stars/jahre-instagram-posts-stars-gingen-decke-34927458`

**Comment by okiehsch** [comment]
I can't reproduce, the video stops for 2-3 seconds and then ot plays without issue.

**Comment by mapx-** [comment]
I can reproduce the issue in FF:
- close the tab, delete cache / cookies, new tab running the video
- video starts then stops and after 2-3 seconds some ads appear then the video continues running well (no `media` type in this first test)
- if I refresh the page the `media` type is correctly redirected and the video is running well

**Comment by okiehsch** [comment]
I can reproduce following your steps.
Seems like something is broken on their site as well. 
If I remove our filters and follow your steps, meaning the site should show regular pre roll ads, I get the same
"video starts then stops and after 2-3 seconds some ad images appear then the video wothiut ads  continues running"´
result and after refresh I get the usualpre roll ads.

**Comment by mapx-** [comment]
test these filters (no delay, no ads) repeating the steps above:
```
@@*$xhr,3p,domain=web.de
*$media,redirect=noop-0.1s.mp3,3p,domain=web.de
```

**Comment by okiehsch** [comment]
````
@@*$xhr,3p,domain=gmx.at|gmx.ch|gmx.net|web.de
*$media,redirect=noop-0.1s.mp3,3p,domain=gmx.at|gmx.ch|gmx.net|web.de
````
to fix all domains.

**Comment by akks1981** [comment]
> 
> 
> ```
> @@*$xhr,3p,domain=gmx.at|gmx.ch|gmx.net|web.de
> *$media,redirect=noop-0.1s.mp3,3p,domain=gmx.at|gmx.ch|gmx.net|web.de
> ```
> 
> to fix all domains.

with this no video longer works @we.de and gmx.de

**Comment by akks1981** [comment]
> 
> 
> test these filters (no delay, no ads) repeating the steps above:
> 
> ```
> @@*$xhr,3p,domain=web.de
> *$media,redirect=noop-0.1s.mp3,3p,domain=web.de
> ```

for me no video longer works @ web.de and gmx.de
close the tab, delete cache / cookies


test, click on play and look:


https://web.de/magazine/auto/auto-betanken-tankdeckel-33168314


https://www.gmx.net/magazine/unterhaltung/stars/jahre-instagram-posts-stars-gingen-decke-34927458

**Comment by mapx-** [comment]
all fine on my end

**Comment by akks1981** [comment]
> 
> 
> all fine on my end

witch firefox ver u use? i use 80b1 with uBlock Origin v1.28.5b9

**Comment by okiehsch** [comment]
It's a CNAME issue, that is caused by 
`*$media,redirect=noop-0.1s.mp3,3p,domain=web.de`
Does 
`@@||akamaiedge.net^$media,domain=web.de`
work for you?

**Comment by okiehsch** [comment]
That filter will only work for your `web.de-link`.

**Comment by akks1981** [comment]
> @@||akamaiedge.net^$media,domain=web.de

web de works thx

**Comment by okiehsch** [comment]
Wait 5-10 minutes then update the `uBO-filters` list, gmx should also work after that.

**Comment by akks1981** [comment]
thx

**Comment by mapx-** [comment]
maybe we should use `redirect-rule` and not `redirect`

**Comment by okiehsch** [comment]
We have to keep the damoh-media block for those sites those requests are not blocked on any list as far as I can see.
What media request get's redirected on your end at
`https://www.gmx.net/magazine/unterhaltung/stars/jahre-instagram-posts-stars-gingen-decke-34927458`
?

**Comment by okiehsch** [comment]
It originates from `flashtalking.com` on my end.

**Comment by mapx-** [comment]
- tbh I got a media 3p type request only disabling all scripts, it was some `advision` request (already blocked by EL) => the reason to add the redirecting filter, otherwise the first filter was enough
- testing your link I get no video ad (3p media request)

**Comment by mapx-** [comment]
ok, testing again I get (and redirected)
`https://s2.adform.net/Banners/Elements/Files/139545/8262082.mp4?bv=0&ADFAssetID=8262082`

**Comment by okiehsch** [comment]
Ok, if all are already blocked by filter lists  the `redirect-rule` should be good enough.

**Comment by JobcenterTycoon** [comment]
With the filter `web.de,gmx.*##+js(aeld, timeupdate)` its no longer needed to whitelist `des.smartclip.net` good for users with DNS adblocker

**Comment by okiehsch** [comment]
Can the exception be removed completely?
If you add
````
web.de,gmx.*##+js(aeld, timeupdate)
@@*$xhr,3p,domain=gmx.at|gmx.ch|gmx.net|web.de,badfilter
````
any breakage or pre-roll ads?

**Comment by JobcenterTycoon** [comment]
yes no breakage or pre roll ads for me

**Comment by akks1981** [comment]
New Video ad 

https://www.gmx.net/magazine/unterhaltung/stars/michael-wendler-angst-wendler-aeussert-eklat-35162506

**Comment by mapx-** [comment]
test this filter

`||zomap.de^$xhr,media,domain=player.glomex.com`

**Comment by akks1981** [comment]
> 
> 
> test this filter
> 
> `||zomap.de^$xhr,media,domain=player.glomex.com`

works! thx

**Comment by JobcenterTycoon** [comment]
Video is again broken on `https://www.gmx.net/magazine/politik/wahlen/us-wahl/rudy-giuliani-schlaegt-verbal-35281230#.homepage.pointOfViewTeaser.Giuliani%20schl%C3%A4gt%20verbal%20um%20sich%20-%20dann%20l%C3%A4uft%20ihm%20Farbe%20%C3%BCbers%20Gesicht.0`
